libbeat/cmd/instance/beat.go

```diff
@@ -139,6 +144,27 @@ type beatConfig struct {
 	TimestampPrecision *config.C `config:"timestamp"`
 }
 
+type certReloadConfig struct {
+	tlscommon.Config `config:",inline" yaml:",inline"`
+	Reload           cfgfile.Reload `config:"restart_on_cert_change" yaml:"restart_on_cert_change"`
+}
+
+func (c certReloadConfig) Validate() error {
+	if c.Reload.Period < time.Second {
+		return errors.New("'restart_on_cert_change.period' must be equal or greather than 1s")
+	}
```

Comment:
We should also add the platform validation here.

```suggestion
	if c.Reload.Period < time.Second {
		return errors.New("'restart_on_cert_change.period' must be equal or greather than 1s")
	}
	if c.Reload.Enabled && runtime.GOOS == "windows" {
		return errors.New("'restart_on_cert_change' is not supported on Windows")
	}
```

Reply:
To accept the suggestion you need a new import `runtime`.

Reply:
I ended up adding both your changes on 562a476c37. Only when committing  the changes I realised `runtime` is already imported on this file.

Reply:
Thanks a lot for already suggesting the validation code :heart: